src/classes/XmlFile.ts

```diff
@@ -87,3 +90,13 @@ export class UnhandledXmlFile extends XmlFile {
 		return new UnhandledXmlFile(location, await archive.readText(location));
 	}
 }
+
+export class XmlFileWithContentTypes extends XmlFileBase {
```

Comment:
Oh, you found a mistake that I made but the class is still needed. `fromArchiveComplex` was supposed to be just `fromArchive`. That method that differentiates `XmlFileWithContentTypes` from `XmlFile`. `fromArchive` takes an additional `_contentTypes` argument

Reply:
I see that sine classes extend from `XmlFileWithContentTypes`, and others extend from `XmlFile` -- what's the pattern? It seems unwise, at a glance, to have different contracts for `.fromArchive` on different classes. I realise I don't have all the information, but I wonder if its not easier to let _all_ classes extend from `XmlFileWithContentTypes`, and then simply leave the 2nd argunent to `.fromArchive` unused in cases where you dont need it.

This would avoid some conceptual overhead that I'm running into myself, and reduces the amount of classes in file `XmlFile` because the abstract class and (proposed implementation of) `XmlFile` can be removed.

~Also, `_contentTypes` is the unique argument in one implementation but its not used?~

Reply:
Thanks for this, I understand a lot more of it now :D